compiler/src/org.graalvm.compiler.asm.aarch64/src/org/graalvm/compiler/asm/aarch64/AArch64Assembler.java

```diff
@@ -3330,44 +3277,2087 @@ public void patch(int codePos, int relative, byte[] code) {
+                                    replicateBit(bitArray[6], 8) << 48 |
+                                    replicateBit(bitArray[5], 8) << 40 |
+                                    replicateBit(bitArray[4], 8) << 32 |
+                                    replicateBit(bitArray[3], 8) << 24 |
+                                    replicateBit(bitArray[2], 8) << 16 |
+                                    replicateBit(bitArray[1], 8) << 8 |
+                                    replicateBit(bitArray[0], 8);
+                    registerImmediate(immediateMap, imm64, imm8, 7, BitValues.ZERO, BitValues.ONE);
+                    /* cmode<0> == 1 && op == 0 */
+                    long imm32 = bitArray[7] << 31 | notBit(bitArray[6]) << 30 | replicateBit(bitArray[6], 5) << 25 | (imm8 & 0x3F) << 19;
+                    imm64 = imm32 << 32 | imm32;
+                    registerImmediate(immediateMap, imm64, imm8, 7, BitValues.ONE, BitValues.ZERO);
+                    /* cmode<0> == 1 && op == 1 */
+                    imm64 = bitArray[7] << 63 | notBit(bitArray[6]) << 62 | replicateBit(bitArray[6], 8) << 54 | (imm8 & 0x3F) << 48;
+                    registerImmediate(immediateMap, imm64, imm8, 7, BitValues.ONE, BitValues.ONE);
+                }
+
+                ImmediateEncodings[] table = immediateMap.values().toArray(new ImmediateEncodings[0]);
+                Arrays.sort(table);
+
+                return table;
+            }
+
+            /*
+             * Contains the encodings associated with each 64-bit immediate value. Since multiple
+             * cmode:op combinations can be used to represent the same value, each possible encoding
+             * combination must be recorded.
+             */
+            private static final class ImmediateEncodings implements Comparable<ImmediateEncodings> {
+
+                public final long imm;
+
+                /* All of these operations are indexed by the bits cmode<3:0>:op. */
+                private final boolean[] validEncoding;
+                private final byte[] imm8;
+
+                private ImmediateEncodings(long imm) {
+                    this.imm = imm;
+                    this.validEncoding = null;
+                    this.imm8 = null;
+                }
+
+                public static ImmediateEncodings createRepresentativeEncoding(long imm) {
+                    return new ImmediateEncodings(imm);
+                }
+
+                protected ImmediateEncodings(long imm, byte imm8, int cmodeBits3to1, BitValues cmodeBit0, BitValues op) {
+                    this.imm = imm;
+                    this.validEncoding = new boolean[32];
+                    this.imm8 = new byte[32];
+                    for (int bit0 : cmodeBit0.values) {
+                        for (int opBit : op.values) {
+                            int cmodeOpEncoding = getCmodeOpEncoding(cmodeBits3to1, bit0, opBit);
+                            assert !validEncoding[cmodeOpEncoding];
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Use of identifier that is a keyword in later versions of Java (NM_FUTURE_KEYWORD_USED_AS_IDENTIFIER)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-use-of-identifier-that-is-a-keyword-in-later-versions-of-java-nm-future-keyword-used-as-identifier).
The identifier is a word that is reserved as a keyword in later versions of Java, and your code will need to be changed in order to compile it in later versions of Java.